### PR TITLE
fix: UI: Refactoring @atlasmap/core services

### PR DESCRIPTION
**Pull request by igarashitm**
fix: UI: Let initialization-service focus on initialization
Fixes: #2879
Relocated document/mapping/file handling from initialization service to the other services.

fix: Fixed wording of catalog/digest/ADM thing
Fixes: #2724

fix: Relocated digest file handling into file-management service(#2726). Also added a contract of ADMDigest file under contracts/ (#2878).
Fixes: #2726

fix: Split file update task from user ADM export
Fixes: #2899

fix: UI: serializer adds scope property to all fields
Fixes: #2900

fix: Pushed some of the logic back from @atlasmap/atlasmap utils to @atlasmap/core (#2883)

fix: UI: Don't use observable for a single event
Fixes: #2915

fix: Encapsulate document specific inspection handling into each model classes
Fixes: #2917

### CODE REVIEW

File: ui/packages/atlasmap-core/src/models/config.model.ts [hunk 1]
```diff
@@ -204,9 +199,12 @@ export class ConfigModel {
             '&collectionClassName=' + docInitModel.collectionClassName;
         }
       }
+    } else {
+      docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
     }
 
-    if (docInitModel.isSource) {
+    // somehow docInitModel.isSource gets always truthy... so explicitly compare with true
+    if (docInitModel.isSource === true) {
```
Comment (pleacu) [inline ui/packages/atlasmap-core/src/models/config.model.ts:207]: That's strange - so if `docInitModel.isSource` is false it's still true?

Comment (igarashitm) [reply ui/packages/atlasmap-core/src/models/config.model.ts:207]: yep, not sure what caused it, as if the type (boolean) is lost at this point and recognized as string

File: ui/packages/atlasmap-core/src/models/inspect/xml-inspection.model.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
+/*
```
Comment (pleacu) [inline ui/packages/atlasmap-core/src/models/inspect/xml-inspection.model.ts:1]: I like the models/inspect/... architecture - much easier to find.

File: ui/packages/atlasmap-core/src/services/field-action.service.ts [hunk 3]
```diff
@@ -49,13 +48,10 @@ export class FieldActionService {
...
   async fetchFieldActions(): Promise<boolean> {
-    return new Promise<boolean>((resolve, reject) => {
+    return new Promise<boolean>((resolve) => {
       if (this.cfg.preloadedFieldActionMetadata) {
         this.clearActionDefinitions();
-        if (
-          this.cfg.preloadedFieldActionMetadata &&
-          this.cfg.preloadedFieldActionMetadata.ActionDetails
-        ) {
+        if (this.cfg.preloadedFieldActionMetadata.ActionDetails) {
```
Comment (pleacu) [inline ui/packages/atlasmap-core/src/services/field-action.service.ts:54]: Does this want to be 'if (this.cfg.preloadedFieldActionMetadata?.ActionDetails) {'

Comment (igarashitm) [reply ui/packages/atlasmap-core/src/services/field-action.service.ts:54]: It is already checked at L52 I think?

Comment (pleacu) [reply ui/packages/atlasmap-core/src/services/field-action.service.ts:54]: yup - missed that

File: ui/packages/atlasmap-core/src/models/config.model.ts (hunk 1 above)
Comment (igarashitm) [reply ui/packages/atlasmap-core/src/models/config.model.ts:207]: @gashcrumb @riccardo-forina do you know anything about this one? `docInitModel.isSource` is a typed boolean property and it is working fine on master, but after this refactoring `if(docInitModel.isSource)` evaluates to truthy even if it's false.

**Review by pleacu — APPROVED** [review]
This looks good to me - I'd say push it asap and we can deal with any issues.

**Comment by igarashitm** [comment]
@pleacu thanks! I'll debug a bit more and make sure at least import/export path works.

File: ui/packages/atlasmap-core/src/models/config.model.ts (hunk 1 above)
Comment (igarashitm) [reply ui/packages/atlasmap-core/src/models/config.model.ts]: nevermind, turned out it's caused by the type cast from JSON.parse() outcome into the contract interface. If the value in the JSON is string (e.g. {"isSource": "true"}) then a string value is delivered through a boolean property. It had to be cast to `string` anyway. crappy type system.

**Pull request merged**